browser/custom_app_icons/android/BUILD.gn

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) 2025 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import("//build/config/android/rules.gni")
+import("//build/config/locales.gni")
+import("//chrome/common/features.gni")
+import("//tools/grit/grit_rule.gni")
+
+android_library("java") {
+  sources = [
+    "java/src/org/chromium/brave/browser/custom_app_icons/CustomAppIconsEnum.java",
+    "java/src/org/chromium/brave/browser/custom_app_icons/CustomAppIconsManager.java",
+    "java/src/org/chromium/brave/browser/custom_app_icons/settings/CustomAppIconsAdapter.java",
+    "java/src/org/chromium/brave/browser/custom_app_icons/settings/CustomAppIconsFragment.java",
+    "java/src/org/chromium/brave/browser/custom_app_icons/settings/CustomAppIconsListener.java",
+  ]
+  deps = [
+    ":java_resources",
+    "//base:tasks_java",
+    "//brave/browser/utils/android:java",
+    "//chrome/android:chrome_base_module_resources",  # adding dep for
+                                                      # ic_launcher &
+                                                      # themed_app_icon
+    "//chrome/browser/preferences:java",
+    "//third_party/androidx:androidx_annotation_annotation_java",
+    "//third_party/androidx:androidx_fragment_fragment_java",
+    "//third_party/androidx:androidx_preference_preference_java",
+    "//third_party/androidx:androidx_recyclerview_recyclerview_java",
+  ]
+
+  resources_package = "org.chromium.brave.browser.custom_app_icons"
+}
+
+android_resources("java_resources") {
+  sources = []
+  foreach(density,
```

Comment:
@bridiver I have iterated through all densities in this context to ensure no duplicates.